### PR TITLE
feat: Add Nix flake

### LINKED ISSUE #25: [FEATURE REQUEST] Standalone CLI
I just got a first look at this and it looks like it'll be a lot easier for me to work with than AGS. 

There is a major problem with this, though. It's only a module, it doesn't work as a standalone Python application. Even the CLI mentioned in the wiki requires invoking this as a module using python -m.

Now, this doesn't seem too big of a problem with most Linux distributions or package managers...

...until you start working with it on Nix or NixOS. Then this "only a module" approach is a major major headache, because by design Nix does _not_ support simply sticking a library in globally and expecting it to work. Many Python modules can still work this way with Nix.

However, because this project uses GObject Introspection, you can't install the module globally and have it work, as a lot of environment variables GIS sets up are not going to be there. Nix documentation and pretty much every guide in the universe for Nix/NixOS would say to make a python module part of a Nix shell... but that's not really practical for a widget system like this where you're not going to be devving with it 99% of the time but having the module running like an application.

End result: Just installing the module and importing fabric results in it complaining about not finding the Gtk namespace because the environment it needs isn't there and won't be there the way it will be in other Linux environments. The only way I can see to get your package to work on NixOS involves a lot of work involving creating a package for your module (As it is not in nixpkgs.) and then create another package that depends on that first package to contain the configuration I'd want to use. 

It's a lot of work that could be sidestepped by there simply being an executable Python script available in this package that Nix would be able to wrap properly. Can't wrap a module, _can_ wrap an application. 

I'll understand if you WONTFIX on this one since you're probably more interested in supporting Arch, but I imagine this might be a trivial chance if all the script really needs to do is import fabric and run the command line as if it was "python -m fabric," yes?

### PR DESCRIPTION
**Pull request by wholikeel**
Enables building and running on systems using the Nix package manager.

Tested with all examples within the repository.

Also 9b5da59 resolves #25 (I use NixOS as well). People will most likely have a better experience 
by building via [buildPythonApplication](https://nixos.org/manual/nixpkgs/stable/#buildpythonapplication-function) and adding this flake's python package as a dependency.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,72 +1,63 @@
...
+  outputs =
+    { nixpkgs, utils, ... }:
+    utils.lib.eachDefaultSystem (
+      system:
+      let
+        overlay = final: prev: {
+          pythonPackagesExtensions = prev.pythonPackagesExtensions ++ [
+            (python-final: python-prev: {
+              # `fabric` was taken
+              fabric-widgets = prev.callPackage ./default.nix { };
```
Comment (wholikeel) [inline flake.nix:21]: > [!NOTE]  
> `fabric` was taken; in any attempt to officially package Fabric either as a standalone application or Python package an available name needs to be chosen.

Comment (its-darsh) [reply flake.nix:21]: it's okay to use either `python-fabric` so we have a uniform with the AUR package or `fabric-widgets`

Comment (wholikeel) [reply flake.nix]: > it's okay to use either `python-fabric`

Changed in f11ff86 :)

**Comment by wholikeel** [comment]
This is completed; it's just up to @its-darsh whether or not he wants to support Nix. I'll understand if he doesn't since he is not a Nix user.

**Comment by its-darsh** [comment]
@wholikeel i don't think merging without testing is a good idea, i'm looking for testers atm

**Comment by wholikeel** [comment]
I wrote an [example/template](https://github.com/wholikeel/fabric-nix-example), should help Nix users unfamiliar with `buildPythonApplication`.

**Comment by wholikeel** [comment]
This has gotten quiet verbose and I feel bad polluting the project with various Nix files when you don't use Nix. I've extracted the nix files to a different [repo](https://github.com/wholikeel/fabric-nix) that fetches this one. 

You can either close this PR or I can port this over to here.

**Comment by jasper-clarke** [comment]
Since this commit has been merged it is still unclear to me how to actually employ this in my own configuration.
Could we have a proper update to the documentation instead of this unclear PR?
@wholikeel @its-darsh

**Comment by RoccoRakete** [comment]
I get this when trying to install fabric on NixOS : `error: python-fabric should use `buildPythonPackage` or `toPythonModule` if it is to be part of the Python packages set.`

**Pull request merged**